### ISSUE TITLE
[package] ghc-filesystem/1.5.8: incorrect  name of [pakcage]config.cmake on linux and windows

### ISSUE THREAD

**Issue by sxsong1207** [conan-io/conan-center-index#9357]
The config.cmake file generated on Linux and Windows have different names, which is supposed to be consistent.
The package ghc-filesystem/1.5.8 generates **FindghcFilesystem.cmake** on Linux, but generates **Findghc-filesystem.cmake** on windows, which makes the cmake failed to find the package.

The issue is supposed to be minor.

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version:  name
  * Operating System+version: **Linux Ubuntu 20.04** and **Win10 x64**
  * Compiler+version: **GCC 9.3** + **Visual Studio 19 **
  * Conan version: **conan 1.45.0**

### Steps to reproduce (Include if Applicable)
no specific reproduce.

**Comment by SpaceIm** [comment]
Steps to reproduce is important. Basically I don't reproduce this issue.

**Comment by SpaceIm** [comment]
I'm a bit confused.
First you said that `FindghcFilesystem.cmake` was generated on Linux, and `Findghc-filesystem.cmake` on Windows. And you say it's `Findghc_filesystem.cmake` on Windows.

What I can say is:
- upstream author has changed config file names as of 1.5.2 (`ghcFilesystem-config.cmake` to `ghc_filesystem-config.cmake`)
- since conan-center mimics upstream config files, we have followed this change (even if it's bad to break an interface, but it was library author choice), therefore I've fixed the config file name for versions greater or equal than 1.5.2 in https://github.com/conan-io/conan-center-index/pull/9238

Now I guess that on Linux, you have the old revision of `ghc-filesystem` recipe, still generating `FindghcFilesystem.cmake` for 1.5.8. But on Windows you have the latest revision of this recipe. Use the latest revision on both OS, and you won't need anymore this branching in your CMakeLists, it will be `find_package(ghc_filesystem REQUIRED)`.
